src/components/PopupWithConfirmation.js

```diff
@@ -0,0 +1,44 @@
+/* -------------------------------------------------------------------------- */
+/*                                   imports                                  */
+/* -------------------------------------------------------------------------- */
+import Popup from "./Popup.js";
+
+/* -------------------------------------------------------------------------- */
+/*                         PopupWithConfirmation class                        */
+/* -------------------------------------------------------------------------- */
+
+export default class PopupWithConfirmation extends Popup {
+  constructor({ popupSelector }) {
+    super({ popupSelector });
+    this._popupSelector = document.querySelector(popupSelector);
```

Comment:
`    this._popupSelector = document.querySelector(popupSelector);
`

You don't need this


**No** need to find the popup again in the inheritors of class `Popup` because the parent has already found it in its constructor and it’s `this._popupElement`. It’s available in any inheritor.